Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -381,4 +462,200 @@ protected override void Dispose(bool disposing)
 
         #endregion
     }
+
+    /// <summary>
+    /// 1D implementation of <see cref="ExchangeBufferBase{T, TIndex}"/>
+    /// </summary>
+    /// <typeparam name="T">The element type.</typeparam>
+    /// <remarks>Members of this class are not thread safe.</remarks>
+    public sealed unsafe class ExchangeBuffer<T> : ExchangeBufferBase<T, Index1>
```

Comment:
Looks good to me 🥇 Maybe, we can also generate these classes automatically?

Reply:
I've done this with the text template as well